### PR TITLE
add a new flag to enforce an effective block gaslimit independent of …

### PR DESCRIPTION
**Pull request by roberto-bayardo**
**Description**

Adds a new flag for the block builder to enforce an effective block gaslimit that is independent of the sysconfig / block header gaslimit.  This allows us to increase gas target without a hardfork, while maintaining the same block gaslimit.

This is defined as a miner (aka builder) flag, but is also enforced by the txpool to avoid letting in a transaction requiring gas that exceeds it.

**Tests**

Currently running on all Base mempool and sequencer nodes.

**Additional context**

**Metadata**

### CODE REVIEW

**Review by sebastianst — APPROVED** [review]

File: miner/worker.go [hunk 1]
```diff
@@ -1105,7 +1105,11 @@ func (w *worker) generateWork(genParams *generateParams) *newPayloadResult {
 	}
 	defer work.discard()
 	if work.gasPool == nil {
-		work.gasPool = new(core.GasPool).AddGas(work.header.GasLimit)
+		gasLimit := w.config.EffectiveGasCeil
+		if gasLimit == 0 || gasLimit > work.header.GasLimit {
+			gasLimit = work.header.GasLimit
```
Comment (atenjin) [inline miner/worker.go:1110]: Question: If user set a more higher value in `w.config.EffectiveGasCeil`, then in line 1110, the gasLimit will set back to the old value from header.
For example, if user set 60M for this parameter, while the value in header is 30M, after line 1109 and 1110, the gasLimit is still 30M. 
Is this what we expect?

Comment (roberto-bayardo) [reply miner/worker.go:1110]: yes, this was my intention, to always respect the 1559-configured gaslimit as an upperbound.

Comment (atenjin) [reply miner/worker.go:1110]: emmm? so `EffectiveGasCeil` is useless if set more then block gas limit? So how do you use this value? For example, if the block limit is 30M, then how to set this?

Comment (roberto-bayardo) [reply miner/worker.go:1110]: I'm not going to hand-hold you through this sorry.

**Pull request merged**